classes/Api/Configuration.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Claromentis\ThankYou\Api;
+
+use Claromentis\Core\Config\Config;
+use Claromentis\Core\Config\ConfigDialog;
+use Claromentis\Core\Config\WritableConfig;
+
+// TODO: Move to Configuration directory
+class Configuration
+{
+	private $config_options;
+
+	/**
+	 * Configuration constructor.
+	 *
+	 * @param $config_options
+	 */
+	public function __construct(\Claromentis\ThankYou\Configuration\Configuration $config_options)
+	{
+		$this->config_options = $config_options;
+	}
+
+	/**
+	 * @param WritableConfig $config
+	 * @return ConfigDialog
+	 */
+	public function GetConfigDialog(WritableConfig $config): ConfigDialog
+	{
+		$options = $this->config_options->GetOptions();
+
+		foreach ($options as $config_name => $config_options)
+		{
+			$display = (bool) ($config_options['display'] ?? true);
+			if ($display === false)
+			{
+				unset($options[$config_name]);
+			}
+		}
+
+		//TODO: Replace with Factory.
+		return new ConfigDialog($options, $config);
+	}
+
+	/**
+	 * Saves a Writable Config to the Application.
+	 * Although this method is slightly superfluous at present, it follows practises allowing for the objects nature to change later.
+	 *
+	 * @param WritableConfig $config
+	 */
+	public function SaveConfig(WritableConfig $config)
+	{
+		$config->Save();
+	}
+
+	/**
+	 * Determines whether Tags are enabled for Thank Yous.
+	 *
+	 * @param Config $config
+	 * @return bool
+	 */
+	public function IsTagsEnabled(Config $config)
```

Comment:
Discussed and understood. My original intent was to treat the config more as an _entity_, and pass it around. However Chris has convinced me that it would simplify things to instead allow this to be set in the constructor, and let the Config API manage it. Going to make these changes now.

Reply:
Fixed in 37247674789c39bf1874bc0bf234bad80d39d7bc